### PR TITLE
implement docker

### PR DESCRIPTION
**Pull request by Roma1990Melnichuk**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,7 +3,7 @@ target/
 !.mvn/wrapper/maven-wrapper.jar
 !**/src/main/**/target/
 !**/src/test/**/target/
-.env
+.env.temlate
```
Comment (liliia-ponomarenko) [inline .gitignore]: Why do you ignore the template file?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+# Final stage
+FROM openjdk:17-jdk-alpine
+WORKDIR application
+COPY --from=builder application/dependencies/ ./
+COPY --from=builder application/spring-boot-loader/ ./
+COPY --from=builder application/snapshot-dependencies/ ./
+COPY --from=builder application/application/ ./
+ENTRYPOINT ["java", "org.springframework.boot.loader.launch.JarLauncher"]
+EXPOSE 8080
```
Comment (liliia-ponomarenko) [inline Dockerfile]: add one empty line

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -1,6 +1,5 @@
-spring.datasource.url=jdbc:mysql://localhost:3306/book_store
-spring.datasource.username=root
-spring.datasource.password=test
```
Comment (liliia-ponomarenko) [inline src/main/resources/application.properties]: is it possible to run your application without docker? Better return these properties

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      retries: 3
+
+  app:
+    depends_on:
+      - mysqldb
+    restart: on-failure
+    build: .
+    image: books-service
+    env_file:
+      - .env.temlate
```
Comment (liliia-ponomarenko) [inline docker-compose.yml]: Let’s  think how you will describe local launch in README

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s fix a few comments ;)

File: .gitignore (hunk 1 above)
Comment (Elena-Bruyako) [reply .gitignore]: fix comment
you should ignore env file, not env.template

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comment

File: .gitignore (hunk 1 above)
Comment (Elena-Bruyako) [reply .gitignore]: remove `.env.temlate`

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comment

**Review by liliia-ponomarenko — APPROVED** [review]
Great 😊

**Pull request merged**